native/dlaqps.go

```diff
@@ -0,0 +1,208 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dlaqps computes a step of QR factorization with column pivoting
+// of an m×n matrix A by using Blas-3.  It tries to factorize
+// NB columns from A starting from the row OFFSET, and updates all
+// of the matrix with Dgemm.
+//
+// In some cases, due to catastrophic cancellations, it cannot
+// factorize nb columns. Hence, the actual number of factorized
+// columns is returned in kb.
+//
+// Dlaqp2 computes a QR factorization with column pivoting of the
+// block A[offset:m, 0:n] of the m×n matrix A. The block
+// A[0:offset, 0:n] is accordingly pivoted, but not factorized.
+//
+// On exit, the upper triangle of block A[offset:m, 0:n] is the
+// triangular factor obtained. The elements in block A[offset:m, 0:n]
+// below the diagonal, together with tau, represent the orthogonal
+// matrix Q as a product of elementary reflectors.
+//
+// offset is number of rows of the matrix A that must be pivoted but
+// no factorized. offset must not be negative otherwise Dlaqp2 will panic.
+//
+// On exit, if jpvt[i] = k, then the ith column of A*P was the kth
+// column of A. jpvt must have length n, otherwise Dlapq2 will panic.
+//
+// On exit tau holds the scalar factors of the elementary reflectors.
+// It must have length n, otherwise Dlapq2 will panic.
+//
+// vn1 and vn2 hold the partial and complete column norms respectively.
+// They must have length n, otherwise Dlapq2 will panic.
+//
+// auxv must have length nb, otherwise Dlaqps will panic.
+//
+// f and ldf represent an nb×nb matrix F such that F**T = L*Y**T*A. ldf
+// must be max(1, n), otherwise Dlaqps will panic.
+//
+// Dlaqps is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlaqps(m, n, offset, nb int, a []float64, lda int, jpvt []int, tau, vn1, vn2, auxv, f []float64, ldf int) (kb int) {
+	checkMatrix(m, n, a, lda)
+	checkMatrix(nb, nb, f, ldf)
+	if len(jpvt) != n {
+		panic(badIpiv)
+	}
+	if len(tau) < n {
```

Comment:
At the moment the panic is "lapack: tau has insufficient length". These are part of #228.